LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaPhysical/PhysicalPayload.cs

```diff
@@ -83,9 +83,11 @@ public PhysicalPayload(byte[] _token, PhysicalIdentifier type, byte[] _message)
             {
                 token = _token;
                 identifier = type;
-                message = new byte[_message.Length];
-                Array.Copy(_message, 0, message, 0, _message.Length);
-
+                if (_message != null)

```

Comment:
@Mandur  please review this part too

Reply:
imho, a PULL_RESP packet can't be null (https://github.com/Lora-net/packet_forwarder/blob/d0226eae6e7b6bbaec6117d0d2372bf17819c438/PROTOCOL.TXT#L320). This check might be superficial, but it might be useful in Laurent's lib. So I have nothing against it